### PR TITLE
Add bulk pause/resume commands for deployment schedules

### PR DESCRIPTION
**Pull request by zzstoatzz**
## Summary

This PR adds  flag support to the `prefect deployment schedule pause/resume` commands, enabling bulk operations on all deployment schedules at once.

## Use Cases

- **Maintenance Windows**: Quickly pause all scheduled deployments during system maintenance
- **Troubleshooting**: Stop all scheduled runs while investigating issues
- **Batch Operations**: Resume all schedules after maintenance or testing

## Changes

- Modified `pause_schedule` and `resume_schedule` commands in `src/prefect/cli/deployment.py`
- Added `--all` flag to both commands for bulk operations
- Arguments are now optional to support both single and bulk operations
- Added comprehensive tests for all new functionality

## Examples

```bash
# Pause all deployment schedules
prefect deployment schedule pause --all

# Resume all deployment schedules  
prefect deployment schedule resume --all

# Original single-schedule commands still work
prefect deployment schedule pause my-flow/my-deployment abc123-...
prefect deployment schedule resume my-flow/my-deployment abc123-...
```

## Testing

Added 8 new tests covering:
- Bulk pause/resume operations with multiple deployments
- Edge cases (no deployments, no active/inactive schedules)
- Error handling for conflicting arguments
- Backward compatibility with existing single-schedule commands

All tests passing.

### CODE REVIEW

File: docs/v3/how-to-guides/deployments/manage-schedules.mdx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+```bash
+prefect deployment schedule pause --all
+```
+
+This command will:
+1. Query all deployments in your workspace
+2. Count the number of active schedules
+3. Prompt for confirmation in interactive mode (bypass with `prefect --no-prompt`)
+4. Pause each active schedule
+5. Display progress as it pauses each deployment's schedules
```
Comment (desertaxle) [inline docs/v3/how-to-guides/deployments/manage-schedules.mdx]: This is probably too much inside baseball

File: docs/v3/how-to-guides/deployments/manage-schedules.mdx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+```bash
+prefect deployment schedule resume --all
+```
+
+This command will:
+1. Query all deployments in your workspace
+2. Count the number of inactive schedules
+3. Prompt for confirmation in interactive mode
+4. Resume each paused schedule
+5. Display progress as it resumes each deployment's schedules
```
Comment (desertaxle) [inline docs/v3/how-to-guides/deployments/manage-schedules.mdx]: Also probably too much info

File: src/prefect/cli/deployment.py [hunk 3]
```diff
@@ -501,64 +506,155 @@ async def delete_schedule(
...
+            )
 
-        try:
-            schedule = [s for s in deployment.schedules if s.id == schedule_id][0]
-        except IndexError:
-            return exit_with_error("Deployment schedule not found!")
+        async with get_client() as client:
+            deployments = await client.read_deployments(
+                deployment_filter=DeploymentFilter(), limit=200
+            )
```
Comment (desertaxle) [inline src/prefect/cli/deployment.py]: Should we paginate to ensure all deployments are upated an not just the first 200?

File: src/prefect/cli/deployment.py [hunk 4]
```diff
@@ -501,64 +506,155 @@ async def delete_schedule(
...
+                            update_tasks.append(
+                                client.update_deployment_schedule(
+                                    deployment.id, schedule.id, active=active
+                                )
+                            )
+                            deployment_names.append(deployment.name)
+
+            # Execute all updates in parallel
+            if update_tasks:
+                await gather(*update_tasks)
```
Comment (desertaxle) [inline src/prefect/cli/deployment.py]: We probably need a limiter here to avoid rate limits or giving the server the hug of death.

File: src/prefect/cli/deployment.py [hunk 5]
```diff
@@ -501,64 +506,155 @@ async def delete_schedule(
...
+            exit_with_success(
+                f"{past_tense.capitalize()} {len(update_tasks)} deployment schedule(s)."
+            )
+
+    else:
+        if deployment_name is None or schedule_id is None:
             return exit_with_error(
-                f"Deployment schedule {schedule_id} is already inactive"
+                "Must provide deployment name and schedule ID, or use --all"
             )
```
Comment (desertaxle) [inline src/prefect/cli/deployment.py]: This feels like it should be an early exit at the top of this function.

File: src/prefect/cli/deployment.py [hunk 6]
```diff
@@ -501,64 +506,155 @@ async def delete_schedule(
...
-    async with get_client() as client:
-        try:
-            deployment = await client.read_deployment_by_name(deployment_name)
-        except ObjectNotFound:
-            return exit_with_error(f"Deployment {deployment_name!r} not found!")
+    if _all:
+        if deployment_name is not None or schedule_id is not None:
+            return exit_with_error(
+                "Cannot specify deployment name or schedule ID with --all"
+            )
```
Comment (desertaxle) [inline src/prefect/cli/deployment.py]: This feels like it shouldn't be nested and should be an early exit at the top of this function.

**Review by desertaxle — CHANGES_REQUESTED** [review]
